qiskit/transpiler/preset_passmanagers/__init__.py

```diff
@@ -205,6 +207,16 @@ def generate_preset_pass_manager(
         ValueError: if an invalid value for ``optimization_level`` is passed in.
     """
 
+    # Handle positional arguments for target and backend. This enables the usage
+    # pattern `generate_preset_pass_manager(backend.target)` to generate a default
+    # pass manager for a given target.
+    if isinstance(optimization_level, Target):
```

Comment:
Personally, I'd rather this function not support any positional arguments. 

I think we want a more ergonomic API on top of this function eventually anyways, e.g. https://github.com/Qiskit/qiskit/pull/12161, so it seems too kludgy to me to try and make this accept a target positionally here.

Reply:
#12161 is actually what I'm not sure we want to do. Adding a 3rd API with different semantics to do what we already have two interfaces for seems like a mistake to me. If people really want a different name I feel like we really should just alias it (but I still don't think it's worth it). But adding yet another entrypoint to accomplish the same thing feels like a mistake. If people are complaining about the ergonomics of the existing interface I feel like we should just evolve it in-place instead of diverging it again and requiring people to learn yet another thing.

Reply:
Yeah, it's unfortunate that we've ended up here, but I think the right thing to do for users is to make it so the common patterns `{transpile,generate_preset_pass_manager}({backend,target}, optimization_level=2)` work the same in both forms.  It ends up in an ugly signature for us, but we can tidy that up in place and potentially fix the signature properly for Qiskit 2.0+.